moban/repo.py

```diff
@@ -24,11 +24,20 @@ def git_clone(requires):
         if os.path.exists(local_repo_folder):
             reporter.report_git_pull(repo_name)
             repo = Repo(local_repo_folder)
-            repo.git.pull()
-            if require.reference:
-                repo.git.checkout(require.reference)
-            elif require.branch:
-                repo.git.checkout(require.branch)
+            for remote in repo.remotes:
+                if require.git_url in remote.urls:
+                    break
+            else:
+                reporter.report_info_message("checking out new remote")
+                remote = Remote.create(repo, "current", require.git_url)
```

Comment:
second time switching remotes causes:

'fatal: remote current already exists.'

Reply:
Trying to create stable remote names for each remote is going to be difficult - and it should be done with enhancements in git-url-parse to allow matching similar remote names, e.g. discard unnecessary `.git` suffixes.

In order to fix this quickly, I propose doing a hash of the remote url.